### PR TITLE
Feature/bbb 833 create permissions

### PR DESCRIPTION
**Pull request by xytrams**

### CODE REVIEW

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java [hunk 1]
```diff
@@ -85,26 +99,51 @@ public String deleteUser(
...
   }
+
+  @ModelAttribute("permissionsOptions")
+  public List<ReferenceData> permissionsOptions() {
+    ReferenceData admin =
+        new ReferenceData()
+            .description(
+                messageSource.getMessage(
+                    "label.user.permissions.administrator", null, Locale.ENGLISH))
+            .shortCode(Role.LA_ADMIN.name());
```
Comment (stephen-bealine) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java]: Message lookups should be handle by the view, this needs to be consistent with all other controllers/views. The locale is hard coded here, so if/when we introduce Welsh we would have a problem

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java [hunk 2]
```diff
@@ -25,41 +33,47 @@
...
 
-  private UserDetailsFormRequestToUser userDetailsFormRequestToUser;
+  private final UserService userService;
+  private final UserFormRequestToUser userConverter;
+  private final MessageSource messageSource;
 
   @Autowired
   public UserDetailsController(
-      UserService userService, UserDetailsFormRequestToUser userDetailsFormRequestToUser) {
+      UserService userService, UserFormRequestToUser userConverter, MessageSource messageSource) {
```
Comment (stephen-bealine) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java]: Message source not needed. Use thymeleaf lookups within the view

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+  }
+
+  @ModelAttribute("permissionsOptions")
+  public List<ReferenceData> permissionsOptions() {
+    ReferenceData admin =
+        new ReferenceData()
+            .description(
+                messageSource.getMessage(
+                    "label.user.permissions.administrator", null, Locale.ENGLISH))
```
Comment (stephen-bealine) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java]: See other comments regarding this

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  @Autowired
+  public CreateUserController(
+      UserService userService,
+      UserFormRequestToUser userConverter,
+      SecurityUtils securityUtils,
+      MessageSource messageSource) {
+    this.userService = userService;
+    this.userConverter = userConverter;
+    this.securityUtils = securityUtils;
+    this.messageSource = messageSource;
```
Comment (stephen-bealine) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java]: See other comments regarding this

File: acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/steps/SiteSteps.java [hunk 5]
```diff
@@ -564,4 +571,20 @@ public void executeDeleteUsersDBScript() throws SQLException {
     DbUtils db = new DbUtils(settings());
     db.runScript("scripts/delete_users.sql");
   }
+
+  @Then("^I should see the newly created user's permission as \"([^\"]*)\"$")
+  public void iShouldSeeTheNewlyCreatedUserSPermissionAs(String permission) throws Throwable {
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/steps/SiteSteps.java]: This should be within User Management Steps

File: acceptance-tests/src/test/resources/features/site/manage_users.feature [hunk 6]
```diff
@@ -125,3 +106,81 @@ Feature: Dft BlueBadge LA Manage Users
...
+    And I can click Sign in button
+    Then I should see the page titled "Manage users - GOV.UK Manage Blue Badges"
+    And I can click on the "Manage users" link on left navigation
+    When I can click on the "Create a new user" button
+    And I select an option "roleName.LA_ADMIN"
+    And I enter full name and email address and clicks on create a new user button
+    When I search for newly create user using email address
+    Then I should see the newly created user's permission as "Administrator"
+
+  Scenario: Verify creating View only user
```
Comment (stephen-bealine) [inline acceptance-tests/src/test/resources/features/site/manage_users.feature]: Check title? Says view only user but code says LA Editor

File: acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/steps/SiteSteps.java (hunk 5 above)
Comment (sampathmahavithana) [reply acceptance-tests/src/test/java/uk/gov/service/bluebadge/test/acceptance/steps/SiteSteps.java]: As we discussed will move them to separate step definition file in a future sprint. Thanks!

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java (hunk 1 above)
Comment (xytrams) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/UserDetailsController.java]: fixed

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java (hunk 4 above)
Comment (xytrams) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java]: fixed

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java (hunk 3 above)
Comment (xytrams) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/CreateUserController.java]: fixed

**Pull request merged**